python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -60,6 +60,7 @@ def asset(
     key_prefix: Optional[CoercibleToAssetKeyPrefix] = None,
     ins: Optional[Mapping[str, AssetIn]] = ...,
     non_argument_deps: Optional[Union[Set[AssetKey], Set[str]]] = ...,
+    upstream_assets: Optional[Set[CoercibleToAssetKey]] = ...,
```

Comment:
Discussion: How should we handle partition mappings?

Reply:
Some ideas:
* Introduce `UpstreamAsset` class with a partition mapping attribute the user can set - ie `upstream_assets={UpstreamAsset(key="my_upstream_asset", partition_mapping=TimeWindowPartitionMapping(...))}`
* Some kind of class method on `AssetsDefinition` that returns an internal type - ie `upstream_assets={my_upstream_asset.with_partition_mapping(TimeWindowPartitionMapping(...))}`

Reply:
see https://github.com/dagster-io/dagster/pull/14908 for `UpstreamAsset` class that allows for partition mappings

Reply:
another potential partitions option https://github.com/dagster-io/dagster/pull/14912

Reply:
supporting `AssetIn` i think is worth considering, from its API doc block it `Defines an asset dependency.`